### ISSUE TITLE
Possible false positive in `Lint/DuplicateMatchPattern`

### ISSUE THREAD

**Issue by hmcguire-shopify** [rubocop/rubocop#11787]
Given the following case statement:

```ruby
case x
in "a" if false
in "a"
end
```

## Expected behavior

`Lint/DuplicateMatchPattern` should not be reported because of the guard clause

## Actual behavior

```
cop.rb:3:4: W: Lint/DuplicateMatchPattern: Duplicate in pattern detected.
in "a"
   ^^^
```

## Steps to reproduce the problem

```
echo 'case x\nin "a" if false\nin "a"\nend' > fp.rb && bx rubocop fp.rb
```

## RuboCop version

```
❯ bx rubocop -V
1.50.0 (using Parser 3.2.2.0, rubocop-ast 1.28.0, running on ruby 3.2.2) [arm64-darwin22]
  - rubocop-rails 2.19.0
```
